service/webhooks_service.go

```diff
@@ -0,0 +1,100 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/libsv/bitcoin-hc/domains"
+	"github.com/libsv/bitcoin-hc/repository"
+	"github.com/spf13/viper"
+)
+
+// WebhookMaxTries is the maximum number of times a webhook will be retried.
+const WebhookMaxTries = "webhook.maxTries"
+
+// WebhooksService represents Webhooks service and provide access to repositories.
+type WebhooksService struct {
+	repo *repository.Repositories
+}
+
+// GenerateWebhook generates and save new webhook.
+func (s *WebhooksService) GenerateWebhook(url, tHeader, token string) (*domains.Webhook, error) {
+	webhook := domains.CreateWebhook(url, tHeader, token)
+	err := s.repo.Webhooks.AddWebhookToDatabase(webhook)
+	if err != nil {
+		w, _ := s.repo.Webhooks.GetWebhookByUrl(url)
+		if w != nil && !w.Active {
+			err = s.repo.Webhooks.UpdateWebhook(w, w.LastEmitTimestamp, w.LastEmitStatus, 0, true)
+			w.Active = true
+			w.ErrorsCount = 0
+			return w, err
+		}
+		return nil, errors.New("webhook already exists and is active")
```

Comment:
Extract a method from that for example
```suggestion
		s.refreshWebhook(webhook)
```